src/Model/Table/MediaTable.php

```diff
@@ -81,4 +81,48 @@ public function validationDefault(Validator $validator)
 
         return $validator;
     }
+
+    /**
+     * @param Query $query
+     * @param array $options
+     * User choose a category (can be empty) then type search string to search by title and category
+     * if category is empty/default, search by title only.
+     * Return all media found.
+     */
+    public function homeSearch(Query $query, array $options){
```

Comment:
I would move these custom searches into the `MediaController`, rather than the Model.